packages/@glimmer/core/test/utils/tracked-object.js

```diff
@@ -0,0 +1,58 @@
+import { consume, tagFor, dirtyTagFor } from '@glimmer/validator';
```

Comment:
Why not just author this in .ts?

Reply:
It is impossible, you can't have `new TrackedObject` return anything but a `TrackedObject` if you author it as a TS class, and there's no way to tell TS that this `TrackedObject` should get the same keys as the object passed to it. @chriskrycho and I noodled on this for a while with tracked-built-ins, and this was the only way I found you could get the expected behavior.

That aside, Proxy itself is very difficult to type correctly. I haven't been able to figure out a way to type the handlers of a Proxy without casting to `any` multiple times yet. So, I'm not sure TS would help us much with the implementation here.